src/telemetry/telemetry.ts

```diff
@@ -0,0 +1,51 @@
+import * as vscode from "vscode";
+import { UserInfo } from "../clients/sidecar/models/UserInfo";
+import { getTelemetryLogger } from "./telemetryLogger";
+import * as Sentry from "@sentry/node";
+
+/** Given authenticated session and/or userInfo, clean the data & send an Identify event to Segment via TelemetryLogger
+ * @param eventName - The event name to be sent to Segment as a follow up per docs: "follow the Identify call with a Track event that records what caused the user to be identified"
+ * @param userInfo - The UserInfo object from the Authentiation event
+ * @param session - The vscode.AuthenticationSession object from an existing session
+ * ```
+ * sendTelemetryIdentifyEvent({eventName: "Event That Triggered Identify", userInfo: { id: "123", ...} });"
+ * ```
+ */
+export function sendTelemetryIdentifyEvent({
+  eventName,
+  userInfo,
+  session,
+}: {
+  eventName: string;
+  userInfo: UserInfo | undefined;
+  session: vscode.AuthenticationSession | undefined;
+}) {
+  const id = userInfo?.id || session?.account.id;
+  const username = userInfo?.username || session?.account.label;
+  const social_connection = userInfo?.social_connection;
+  let domain: string | undefined;
+  if (username) {
+    //  email is redacted by VSCode TelemetryLogger, but we extract domain for Confluent analytics use
+    const emailRegex = /@[a-zA-Z0-9-]+\.[a-zA-Z0-9-]+/;
```

Comment:
I'd of maybe just done if `@` present, then split and grab `[1]` since we don't care about other properties and expect it to always be an email entry anyway.

Reply:
does sound simpler! I wasn't sure if we were 100% certain this will be email every time. Happy to update if it makes sense to do so in follow ups